models/clm/readme.rst

```diff
@@ -1,40 +1,195 @@
+code, there are a few SourceMods needed to run DART from within CESM. Appropriate 
+SourceMods for each CESM version are available at 
+http://www.image.ucar.edu/pub/DART/CESM
+and should be unpacked into your HOME directory. They will create a ``~/cesm_?_?_?`` 
+directory with the appropriate SourceMods structure. The DART scripting require 
+these SourceMods and expect them to be in your HOME directory.
+
+CLM is a rapidly-moving target and DART is developed and maintained by a small
+group of people. Consequently, we have focused on supporting *released* versions
+of CLM. This documentation and scripting were tested using the CESM 
+tag **release-cesm2.2.0** and CLM tag **release-cesm2.2.01** following the download
+instructions from https://github.com/ESCOMP/CESM .
+
+CLM-DART has been used to assimilate snow data, soil moisture, leaf area index, 
+biomass, solar-induced fluorescence and more. See the `References`_ section below
+for some examples. The list is by no means complete. Each experiment has extended
+the capabilities of the CLM-DART system. The observations supported by
+DART are contained in the *obs_def_land_mod.f90*, *obs_def_tower_mod.f90*, 
+and *obs_def_COSMOS_mod.f90* modules - each of which may be easily extended. 
+New observation types are usually easily supported. There are **many** examples 
+of routines to convert from native data formats (netCDF, HDF, csv, etc.) 
+to a DART observation sequence file.
+
+
+Important Features
+------------------
+
+Land DA is extremely diverse. The support for Land DA as pertains to CLM-DART
+has some features that need to be described in some detail.
+
+
+CLM indeterminate values
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+CLM variables are encoded as rectangular arrays in the netCDF files. (Thankfully!)
+However, this means that some variables have space for layers that are unused.
+Anything with snow layers, for example. CLM has the *SNLSNO* varible to indicate
+which snow layers are active. The *unused* layers may not have the *_FillValue*
+value, but can have 'indeteminate' values. The :doc:`clm_to_dart` 
+must be run to convert these indeterminate values to *_FillValue* to be 
+interpreted correctly by DART.  After the assimilation is complete, the 
+:doc:`dart_to_clm` must be called to replace the *_FillValue* with whatever
+is originally in that slot. Extra care is taken to handle the special case
+of a *trace* of snow. See the *Discussion of Indeterminate Values* 
+section of :doc:`clm_to_dart` for more details.
+
+
+Model Interpolate - The Forward Operator
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Since the subgridscale components of CLM have no explicit location associated 
+with them, the location of every component in the gridcell is the same as the 
+gridcell itself. The DART forward operators fundamentally rely on 
+interpolating the model state to some arbitrary location. At present, the best we
+can do is to create an area-weighted average of all pieces in the gridcell.
```

Comment:
Prior few sentences are important, but a bit vague.   Because subgrid components in CLM are not spatially explicit, given an explicit sub-grid location of a biomass observation (for example), we cannot create a forward operator that spatially interpolates to that precise location.  Currently, any observation within the domain of the grid cell is considered collocated (identical location) as the weighted grid cell average of the land surface property of interest.   This is clearly sub-optimal because this introduces representation error between the grid cell weighted average and the observation.    For this reason, for some applications it may be helpful to aggregate (average) the observations to the same spatial resolution as the CLM grid.  Alternatively, a nice project would be ........